src/java/org/apache/cassandra/hints/HintsWriter.java

```diff
@@ -271,6 +273,7 @@ public void close() throws IOException
             flushBuffer();
             maybeFsync();
             maybeSkipCache();
+            descriptor.size = position();
```

Comment:
why not to just update it in close() ? Then when this is used in try-with-resources in HintsWriteExecutor, we do not need to change there anything at all.

Reply:
this should be probably in finally though, not sure ... as all above can throw